### DISCUSSION TITLE
Ammunition can sometimes get "locked" on the hotbar

### DISCUSSION THREAD

**Discussion by Norc** [FakeFishGames/Barotrauma#14277] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Randomly, my friend couldn't move some revolver rounds out of their hotbar for almost an hour, saying it was locked. Screenshot from another discord user (https://discord.com/channels/103209684680323072/910489857028128818/1261048709765136485) that encountered the same issue. There's apparently been a few other reports also:

![image](https://github.com/FakeFishGames/Barotrauma/assets/64667579/0cbd4bc1-f55d-4aef-96bc-3a4c66340026)

This was also specifically with revolver rounds.

I'm not sure why an item can ever get locked in the hotbar... this seems like a bug. Sorry I can't be more specific with reproduction steps.

**Note**: The server I was on when this happened used only one module, Backpacks and Waistbelts 2.0, but it's unlikely that was the cause. 

### Reproduction steps

1. start a multiplayer server
2. use revolvers with spare ammunition in a separate slot in the hotbar
3. ????

CUrrent beahvior: Hotbar slot with ammunition can sometime get locked for no discernable reason
Expected: It is moved as normal.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Norc** [comment]
I confirmed that the other user who experienced this issue was also only using a single module, and it was a different one "Real Sonar".

**Comment by BlackMoons** [comment]
Iv had this happen too after finding a revolver round on the ground. It may have been after I switched submarines? I believe 'locked' means its a bullet tagged as non-interactable or something.

**Comment by pavel4win** [comment]
I had a similar problem with wires.
I don't know how to replicate it
the wires have been locked in cabinet.

**Comment by piiktoo** [comment]
hey, i literally just got this problem !!! specifically with the revolver rounds as well...
![Capture3](https://github.com/user-attachments/assets/650cd514-2396-40ba-abe0-1b6f6feee2a1)

i don't know how to get rid of it :(

**Reply by lIllIIlI** [reply]
you can edit your character data file to remove the uninteractable tag from the item meanwhile

**Reply by pavel4win** [reply]
just have to go to merchant and sell it.

**Comment by MKMoose** [comment]
I've encountered this twice with my friend, one with a Rifle Round and one I think was Shotgun Shells. We were playing on our own server (hosted by me) with a few mods of my own (just a few subs and some items, no relevant overrides).

First he found the Round lying around, the other time I gave him a few Shells that I picked up after a fight, both times he reported it was locked in his inventory, but I could grab him and take the items from his inventory without issues, suggesting it might be a client-side issue, or some kind of desync where a client shoots a ranged weapon but it's not registered properly on the server, or the ammunition isn't removed for some reason.

**Comment by A2ur3** [comment]
it seems to be related to #14062

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/fb730d07-f6f9-45a7-8f07-200a7819d2af

it stil hasn't been fixed after 1 year...

**Comment by Regalis11** [comment]
I'm sorry it's taken so long for us to address this - unfortunately we still haven't been able to find a way to reproduce the issue, nor been able to get a hold of e.g. save files or other information that might help us diagnose it. :/

I have a suspicion of what could be causing it though, implemented a fix in our private repo now (coming up in the next update):
> Potentially fixed clients sometimes ending up with locked/non-interactable projectiles in their inventory. Haven't been able to repro this, but I suspect it happens when a client incorrectly predicts that the projectile hit. That makes the client hide the projectile, with the assumption that the server will report the hit and remove the item right after, but if the shot failed server-side, the item would go back in the inventory and remain `HiddenInGame`, which makes it appear locked in the inventory.

**Reply by A2ur3** [reply]
Check the video I sent above. I was able to reproduce the issue the same way
Spawn a large number of husks at the station airlock, then stand below them and keep shooting with a revolver while quickly pressing the hotkey

**Reply by ubertpendragon** [reply]
> I'm sorry it's taken so long for us to address this - unfortunately we still haven't been able to find a way to reproduce the issue, nor been able to get a hold of e.g. save files or other information that might help us diagnose it. :/


It was very consistent in the DLC testing session when using my revolver. Possibly because it was hosted on a dedicated server?

**Comment by Regalis11** [comment]
Should now finally be [addressed in our private repo](https://github.com/FakeFishGames/Barotrauma-development/pull/8575) and the unstable version, and will be included in the next update.
